src/main/scala/com/nec/vectorengine/VeProcessImpl.scala

```diff
@@ -406,15 +410,17 @@ final case class WrappingVeo private (val node: Int,
 
   def freeArgsStack(stack: VeCallArgsStack): Unit = {
     withVeoProc {
-      stackRecords.get(stack.args.address) match {
-        case Some(allocation) =>
-          logger.trace(s"[${handle.address}] Releasing veo_args @ ${stack.args.address}")
-          veo.veo_args_free(stack.args)
-          // Remove only after the free() was successful
-          stackRecords.remove(stack.args.address)
+      stackRecords.synchronized {
```

Comment:
Same question as above, do we need this additional lock?

Reply:
Similar response to above.